lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -136,14 +136,6 @@ def scrub(transcript)
           gsub(%r(("securityCode\\?":{\\?"[\w]+\\?":[\d]+,\\?"value\\?":\\?")[\d]*)i, '\1[FILTERED]')
       end
 
-      private
-
-      def add_credentials(post, options)
-        post[:credential] = {}
-        post[:credential][:clientGuid] = @client_guid
-        post[:credential][:authToken] = @auth_token
-      end
-
       def setup_access_token
```

Comment:
Is the idea that `setup_access_token` is part of creation of a gateway object, allowing for retaining the access_token on the gateway? Or will this result in an extra API request?

Reply:
Yes, this is to enable us to make the exchange for an `access_token` when a gateway object is created and we can store that value to be used for other requests. 

It's going to require an additional API request at the time of creation for that gateway but we will check for the presence of an `access_token` when auth is being setup to avoid making extraneous requests to the gateway.